sdk/nodejs/src/request.ts

```diff
@@ -105,130 +25,27 @@ export async function getEnvironmentConfig(
     )
 }
 
-export async function getAllFeatures(
-    user: DVCPopulatedUser,
-    sdkKey: string,
-    options: DevCycleOptions,
-): Promise<Response> {
-    const baseUrl = `${
-        options.bucketingAPIURI || BUCKETING_URL
-    }${FEATURES_PATH}`
-    const postUrl = baseUrl.concat(
-        options.enableEdgeDB ? EDGE_DB_QUERY_PARAM.concat('true') : '',
-    )
-    return await post(
-        postUrl,
-        {
-            body: JSON.stringify(user),
-            retries: 5,
-        },
-        sdkKey,
-    )
-}
-
-export async function getAllVariables(
-    user: DVCPopulatedUser,
-    sdkKey: string,
-    options: DevCycleOptions,
-): Promise<Response> {
-    const baseUrl = `${
-        options.bucketingAPIURI || BUCKETING_URL
-    }${VARIABLES_PATH}`
-    const postUrl = baseUrl.concat(
-        options.enableEdgeDB ? EDGE_DB_QUERY_PARAM.concat('true') : '',
-    )
-    return await post(
-        postUrl,
-        {
-            headers: { 'Content-Type': 'application/json' },
-            body: JSON.stringify(user),
-            retries: 5,
-        },
-        sdkKey,
-    )
-}
-
-export async function getVariable(
-    user: DVCPopulatedUser,
-    sdkKey: string,
-    variableKey: string,
-    options: DevCycleOptions,
+export async function publishEvents(
+    logger: DVCLogger,
+    sdkKey: string | null,
+    eventsBatch: SDKEventBatchRequestBody,
+    eventsBaseURLOverride?: string,
 ): Promise<Response> {
-    const baseUrl = `${
-        options.bucketingAPIURI || BUCKETING_URL
-    }${VARIABLES_PATH}/${variableKey}`
-    const postUrl = baseUrl.concat(
-        options.enableEdgeDB ? EDGE_DB_QUERY_PARAM.concat('true') : '',
-    )
+    if (!sdkKey) {
+        throw new Error('DevCycle is not yet initialized to publish events.')
```

Comment:
it seems kind of weird to be doing a check like this in the request layer?

also the error message seems like it would be incorrect in most cases?

Reply:
yea not sure, this was in the existing code. I can try and move it.